assets/js/navigation.js

```diff
@@ -309,3 +309,47 @@ jQuery("#menu-menu-top-menu li.menu-item-has-children > ul.sub-menu").ready(func
 		if (e.keyCode == "13") $(this).click();
 	});
 });
+
+/**
+ *
+ * Table of contents "scrollspy", highlight current scroll location
+ *
+ */
+
+( function() {
+	indicateCurrentLocation();
+})();
+
+document.addEventListener('scroll', function() {
+	indicateCurrentLocation();
+}, false);
+
+function indicateCurrentLocation(){
+	if (!document.querySelector("#toc")) return;
+	let toc = document.querySelector("#table-of-contents");
+	let sectionHeadings = document.querySelectorAll(".hale-toc-item");
+	let contents = toc.querySelectorAll("li");
+	for (i=0; i+1<sectionHeadings.length; i++) {
+		let nextPosition = sectionHeadings[i+1].getBoundingClientRect().top;
+		if (nextPosition > 150) break; //we stop counting when the next one is above 150 as we are on the current item
+	}
+
+	// A small bit of code to ensure the last item is always "current" when at the very bottom of the page.
+	const documentHeight = Math.max(
+		document.body.scrollHeight,
+		document.body.offsetHeight,
+		document.documentElement.clientHeight,
+		document.documentElement.scrollHeight,
+		document.documentElement.offsetHeight
+	);
+	const windowHeight = window.innerHeight;
+	const maxScroll = documentHeight - windowHeight;
+
+	if (window.scrollY >= maxScroll) i = sectionHeadings.length - 1;
```

Comment:
This bit of code is what's causing the skip sections - by design.  If you're at the bottom of the page, the last item is always "current".  

This means that if someone clicks on the last item and it is too far down the page to be at the top - it is still highlighted.  

Of course, if there are lots of small sections, then some will be missed.  I thought this approach would deal with the most likely scenario.

Reply:
I have asked Lewis about this behaviour to confirm it is okay.